_posts/blog/2018-10-24/2018-10-24-Escrow-Smart-Contract-Specification-in-OpenBazaar.markdown

```diff
@@ -0,0 +1,110 @@
+
+### Intended Use of the Escrow contract
+
+The Escrow contract will store the escrowed funds and state information for _every_ OpenBazaar trade that is using Ethereum (or ERC20 tokens) as the medium of exchange.
+
+We could have, instead, opted to deploy a new escrow contract for each Ethereum-based trade — thereby siloing escrowed funds from each trade in their own smart contract. However, we think the gas requirements for doing so are cost prohibitive, and we fear that would introduce too much friction into Ethereum-based trades.
+
+OpenBazaar trades that use ETH/ERC20 as the medium of exchange are intended to follow the same protocol as those that use a UTXO-based coin as the medium of exchange — and the escrow smart contract is intended to facilitate that.
+
+#### Funding the Trade
+
+Buyers initiate a trade by creating/storing a `Transaction` struct in the Escrow contract and (simultaneously) funding the transaction by sending ETH (or ERC20 tokens) to the Escrow contract. At this point the transaction is in the `FUNDED` state. While in the `FUNDED` state, the buyer may add more ETH (or ERC20 tokens) to escrow if necessary.
+
+#### Releasing Funds from Escrow
+
+While the transaction is in the `FUNDED` state, the escrowed funds can be released only if:
+
+1. Two of the three participants (buyer, seller, and moderator) agree on how the escrowed funds are to be distributed, or 
+2. An amount of time (`timeoutHourse`) has passed since the last time the buyer added funds to escrow.
+   
+The reasoning behind (2) is that it is very common for buyers to not release funds after they’ve received their goods (this is due more to buyer laziness than malice). In that event, we want to make it easy for the seller to claim the escrowed funds without having to coordinate with a moderator.
+
+Funds released from escrow can be split up and sent to various addresses. However, the receiving addresses _must be_ the addresses of the trade’s buyer, seller, or moderator. To reiterate, funds cannot be sent to an address that is not affiliated with the trade in question, but the escrowed funds can be divided up among the participants in any way — so long as 2-of-3 of the parties agree.
+
+Upon release of funds from escrow, the trade is put into the `RELEASED` state. Once in the `RELEASED` state, trades can no longer be altered. All participants who received some of the escrowed funds are noted in the trade’s Transaction struct (via the `beneficiaries` mapping).
+
+(The `beneficiaries` information will be used later, by other contracts, to determine whether or not a given trade was disputed, refunded, etc).
```

Comment:
"etc" should have a period, `refunded, etc.).` as it's an abbreviation of et cetera.